controllers:
  chassis_controller:
    type: rm_chassis_controllers/OmniController
    # ChassisBase
    publish_rate: 100
    enable_odom_tf: true
    power:
      effort_coeff: 12.0
      vel_coeff: 0.0048
      power_offset: 0
    twist_angular: 0.5233
    timeout: 0.1
    pid_follow: { p: 5.0, i: 0, d: 0.3, i_max: 0.0, i_min: 0.0, antiwindup: true, publish_state: true }
    twist_covariance_diagonal: [ 0.001, 0.001, 0.001, 0.001, 0.001, 0.001 ]

    # OmniController
    modules:
      left_front:
        chassis_frame_vel: [ [ -0.147 , 1. , 0. ],
                             [ 0.147 , 0. , 1. ] ] #[[-y,1,0],[x,0,1]]
        wheel_frame_vel: [ [ -0.7071 , 0.7071 ],
                           [ -0.7071 , -0.7071 ] ] #[[cos,sin],[-sin,cos]] 3pi/4
        joint: left_front_wheel_joint
        <<: &wheel_setting
          direction_component: [ 1. , 0. ]
          wheel_radius: 0.07625
          pid: { p: 0.6, i: 0, d: 0.0, i_max: 0.0, i_min: 0.0, antiwindup: true, publish_state: true }
      right_front:
        chassis_frame_vel: [ [ 0.147 , 1. , 0. ],
                             [ 0.147 , 0. , 1. ] ]
        wheel_frame_vel: [ [ 0.7071 , 0.7071 ],
                           [ -0.7071 , 0.7071 ] ] #pi/4
        joint: right_front_wheel_joint
        <<: *wheel_setting
      left_back:
        chassis_frame_vel: [ [ -0.147 , 1. , 0. ],
                             [ -0.147 , 0. , 1. ] ]
        wheel_frame_vel: [ [ -0.7071 , -0.7071 ],
                           [ 0.7071 , -0.7071 ] ]
        joint: left_back_wheel_joint
        <<: *wheel_setting
      right_back:
        chassis_frame_vel: [ [ 0.147 , 1. , 0. ],
                             [ -0.147 , 0. , 1. ] ]
        wheel_frame_vel: [ [ 0.7071 , -0.7071 ],
                           [ 0.7071 , 0.7071 ] ]
        joint: right_back_wheel_joint
        <<: *wheel_setting
